LanguageFeatures/Records/type_promotion_A01_t03.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Type inference and promotion flows through records in much the 
+/// same way it does for instances of generic classes (which are covariant in 
+/// Dart just like record fields are) and collection literals.
+///
+/// @description Checks type promotion for records. Test records with named and
+/// positional fields
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=records
+
+typedef R1 = (int i, {String s})?;
+typedef R2 = (num?, {Object? o});
+
+main() {
+  R1 r1;
+  if (2 > 1) {
+    r1 = (42, s: "answer");
+  }
+  if (r1 != null) {
+    r1.$0;
+    r1.s;
+  } else {
+    r1?.$0;
+    r1?.s;
+  }
+
+  R2 r2 = (42, o: "answer");
+  if (r2 is (int? n, {String o})) {
+    r2.n?.isOdd;
```

Comment:
```suggestion
    r2.$0?.isOdd;
```
We can't use a name associated with a positional field as a getter on the record.